qiskit/transpiler/basepasses.py

```diff
@@ -88,11 +88,17 @@ def __call__(self, circuit, property_set=None):
         elif result is None:
             result_circuit = circuit.copy()
 
-        if self.property_set["layout"]:
+        layout = self.property_set["layout"]
+        if not layout:
+            layout = self.property_set["original_layout"]
+        final_layout = self.property_set["final_layout"]
+        if not final_layout:
+            final_layout = self.property_set["elision_final_layout"]
```

Comment:
I think they must stay outside of that block because you could have a `final_layout` set without having an (initial) `layout` set - at least in principle.... :-)